### ISSUE TITLE
Rules: enforce compound modifier hyphenation (`well documented` -> `well-documented`)

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#410]
## Summary

Add a new lint rule (and `--fix` auto-fix) to catch missing hyphens in common compound modifiers used before a noun, for example `well documented` -> `well-documented` (use `-`[U+002D]).

## Problem

In prose, unhyphenated compound modifiers can be ambiguous and inconsistent across docs. rumdl currently has no dedicated check that flags and fixes these cases. See the current rule catalog for existing formatting and style checks. 

### Lint

Flag cases where a known compound modifier appears immediately before a noun.

Example (should warn or error, and be fixable):
```markdown
This is a well documented guide.
```

### Auto-fix

Insert `-`[U+002D] between the modifier parts:
```markdown
This is a well-documented guide.
```

## Non-goals

Do not flag predicate usage where the modifier is not directly before a noun:
```markdown
This guide is well documented.
```

## Implementation notes

- Only analyze normal text content (skip fenced code blocks, inline code, and link destinations).
- Add tests covering:
  - Fixable compound modifier before a noun
  - Predicate usage not flagged
  - No changes inside code spans or code blocks
- Document the new rule in the rules reference and include it in the rule metadata export.

## References

- https://github.com/rvben/rumdl/blob/main/docs/RULES.md
- https://github.com/rvben/rumdl/blob/main/docs/global-settings.md
- https://github.com/rvben/rumdl/blob/main/docs/inline-configuration.md

**Comment by rvben** [comment]
Closing as out of scope — see #408 for details.
